instrumentation/http/test/instrumentation/http/patches/client_test.rb

```diff
@@ -16,14 +16,21 @@
 
   before do
     exporter.reset
+    @orig_propagation = OpenTelemetry.propagation
+    propagator = OpenTelemetry::Trace::Propagation::TraceContext.text_map_propagator
+    OpenTelemetry.propagation = propagator
     instrumentation.install({})
     stub_request(:get, 'http://example.com/success').to_return(status: 200)
     stub_request(:post, 'http://example.com/failure').to_return(status: 500)
     stub_request(:get, 'https://example.com/timeout').to_timeout
   end
 
-  # Force re-install of instrumentation
-  after { instrumentation.instance_variable_set(:@installed, false) }
+  after do
+    # Force re-install of instrumentation
+    instrumentation.instance_variable_set(:@installed, false)
```

Comment:
`reset` is only defined for the `InMemorySpanExporter` - it isn't part of the `Exporter` API - arguably that exporter should be moved out of the SDK to the common package (or somewhere else).

The force-reinstall is used in instrumentation tests (only) to avoid state leakage between tests. Re-installing instrumentation is not a supported operation in general.

Reply:
👍🏼